lib/skate/detours/detours.ex

```diff
@@ -65,34 +67,33 @@ defmodule Skate.Detours.Detours do
           past: list(DetailedDetour.t()) | nil
         }
   def grouped_detours(user_id) do
-    detours =
-      list_detours()
-      |> Enum.map(&db_detour_to_detour/1)
-      |> Enum.filter(& &1)
-      |> Enum.group_by(fn
-        %ActivatedDetourDetails{details: %{status: status}} -> status
-        %{status: status} -> status
-      end)
-
     %{
-      active: Map.get(detours, :active, []),
-      draft:
-        detours
-        |> Map.get(:draft, [])
-        |> Enum.filter(fn detour -> detour.author_id == user_id end),
-      past: Map.get(detours, :past, [])
+      active: detours_for_user(user_id, :active),
+      draft: detours_for_user(user_id, :draft),
+      past: detours_for_user(user_id, :past)
     }
   end
 
-  @spec db_detour_to_detour(Detour.t()) :: DetailedDetour.t() | nil
-  @spec db_detour_to_detour(status :: Detour.status(), Detour.t()) :: DetailedDetour.t() | nil
-  def db_detour_to_detour(%{} = db_detour) do
-    db_detour_to_detour(categorize_detour(db_detour), db_detour)
+  def detours_for_user(user_id, status) do
+    Skate.Detours.Db.Detour.Queries.select_detour_list_info()
+    |> apply_user_and_status_filter(user_id, status)
+    |> Repo.all()
+    |> Enum.map(&db_detour_to_detour/1)
+    |> Enum.reject(&is_nil/1)
+  end
+
+  defp apply_user_and_status_filter(query, user_id, :draft) do
+    where(query, [detour: d], d.status == :draft and d.author_id == ^user_id)
+  end
+
+  defp apply_user_and_status_filter(query, _user_id, status) do
+    where(query, [detour: d], d.status == ^status)
```

Comment:
thought(non-blocking): what do you think of putting these in `Skate.Detours.Db.Detour.Queries`? I'm not that well versed in Elixir, so I'll defer to you here, but my thinking is that query building functions would belong in the `Queries` module.